server/src/test/java/org/apache/druid/segment/indexing/granularity/UniformGranularityTest.java

```diff
@@ -358,6 +363,54 @@ public void testUniformGranularitySpecWithLargeNumberOfIntervalsDoesNotBlowUp()
     Assert.assertEquals(3600 * 24 * 365 * 10 + 3 * 24 * 3600, count);
   }
 
+
+  @Test
+  public void testSegmentGranularityEqualToQueryGranularity()
+  {
+    final Granularity secondGranularity = Granularities.SECOND;
+    final Granularity secondDurationGranularity = new DurationGranularity(1000, 0);
+    try {
+      new UniformGranularitySpec(
+          secondGranularity,
+          secondGranularity,
+          Collections.emptyList()
+      );
+      new UniformGranularitySpec(
+          secondGranularity,
+          secondDurationGranularity,
+          Collections.emptyList()
+      );
+    }
+    catch (DruidException e) {
```

Comment:
We can simply validate that the instantiation was successful by asserting that the query and segment granularities are equal in this case.

Or do a `catch (Exception e)`; otherwise, if someone changes the type of exception thrown, this test will still pass. 

I think the fist approach would be cleaner.

Reply:
Done